### PR TITLE
Add draft auto-save feature to Composer

### LINKED ISSUE #92: Add auto-save draft functionality for notes
## Problem

Users lose their work when writing notes if the browser tab closes unexpectedly or the page refreshes. Unlike articles which have comprehensive draft management, notes have no protection against data loss during composition.

## Proposed Solution

Implement a simple auto-save mechanism for notes that:

- Automatically saves note content to local storage while typing
- Restores the last draft when returning to the composer
- Clears the draft after successful posting
- Only maintains one draft at a time (simpler than article drafts)

## Implementation Approach

- Use browser `localStorage` for client-side storage
- Auto-save with debouncing (3–5 second delay after typing stops)
- Show restore prompt when user returns with existing draft
- Include draft timestamp and basic metadata (visibility, language)
- Add manual "clear draft" option

## Benefits

- Prevents frustrating content loss
- Minimal server impact (client-side only)
- Simple UX compared to full draft management
- Consistent with user expectations from other platforms

## Technical Notes

This would modify the `Composer` component to add draft persistence hooks and local storage utilities. The feature should gracefully degrade when `localStorage` is unavailable.

### PR DESCRIPTION
**Pull request by kanghyojun**
Fixes #92

## Summary

This PR implements client-side auto-save functionality for note composition, preventing data loss when users accidentally close their browser tab or experience page refreshes. The implementation uses localStorage and provides a UX for draft recovery.

## Key Features

1. Cached localStorage Availability Check

- Performs a one-time localStorage availability test on page load and caches the result (web/lib/draft-storage.ts:19-34)
- Handles edge cases like Safari private mode gracefully

2. Optimized Auto-Save with 1.5s Debounce

- While the original issue proposed 3-5 seconds, i optimized the debounce to 1.5 seconds after discussion for better UX
- Saves draft state including content, visibility, language, media attachments, and quoted posts
- Only saves when content actually changes (performs deep equality check to avoid unnecessary writes)
- Automatically clears draft when content is deleted or post is successfully published

3. Draft Management

- Restore Draft: Shows restore button when returning to empty composer with existing draft
- Clear Draft: Manual option to discard saved draft with confirmation dialog                                                            
- Auto-Clear: Automatically removes draft after successful post submission
- Visual Feedback: Displays timestamped "Draft saved" indicator for 3 seconds after each save

## Etc
 
- New Files:
  - web/lib/draft-storage.ts - Draft storage utilities with schema validation using Valibot
- Modified Files:
  - web/islands/Composer.tsx - Integrated auto-save logic with existing component lifecycle
  - web/locales/*.json - Added i18n strings for 5 languages (en, ja, ko, zh-CN, zh-TW)

### Auto-save in Action:

https://github.com/user-attachments/assets/7843c738-8ca2-43f8-9752-c335e275a2ff

## Summary by CodeRabbit

* **New Features**
  * Drafts auto-save while composing (debounced) and persist between sessions, including media/visibility/language changes; restore or clear drafts manually; drafts cleared after submission; brief on-screen "draft saved" confirmation with timestamp.

* **Localization**
  * Draft-management UI strings added for supported languages (en, ja, ko, zh-CN, zh-TW).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: web/locales/en.json [hunk 1]
```diff
@@ -352,7 +350,11 @@
     "post": "Post",
     "postFailed": "Failed to submit your note. Please try again later.",
     "uploadImage": "Upload image",
-    "markdownEnabled": "Markdown enabled"
+    "markdownEnabled": "Markdown enabled",
+    "restoreDraft": "Restore Draft",
```
Comment (dahlia) [inline web/locales/en.json]:
```suggestion
    "restoreDraft": "Restore draft",
```

File: web/locales/en.json [hunk 2]
```diff
@@ -352,7 +350,11 @@
...
+    "draftSaved": "Draft saved",
+    "clearDraft": "Clear Draft",
```
Comment (dahlia) [inline web/locales/en.json]:
```suggestion
    "clearDraft": "Clear draft",
```

**Review by dahlia — COMMENTED** [review]
Generally looks good to me!

**Comment by kanghyojun** [comment]
@dahlia When I ran the `deno task check` locally, it seemed fine, but the issue appears to have arisen during the force push.

Should I request a GitHub Action rebuild?
